pkg/acsengine/template_generator.go

```diff
@@ -679,6 +679,12 @@ func (t *TemplateGenerator) getTemplateFuncMap(cs *api.ContainerService) templat
 		"IsNSeriesSKU": func(profile *api.AgentPoolProfile) bool {
 			return isNSeriesSKU(profile)
 		},
+		"UseSinglePlacementGroup": func(profile *api.AgentPoolProfile) bool {
+			return *profile.SinglePlacementGroup
+		},
+		"HasAvailabilityZones": func(profile *api.AgentPoolProfile) bool {
+			return profile.AvailabilityZones != nil && len(profile.AvailabilityZones) > 0
```

Comment:
Let's create a convenience function `hasAvailabilityZones` that this templat func calls...